components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -479,6 +481,11 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         jwtClaimsSetBuilder.notBeforeTime(new Date(curTimeInMillis));
         jwtClaimsSetBuilder.claim(CLIENT_ID, consumerKey);
 
+        if (OAuthServerConfiguration.getInstance().isAddTenantDomainToAccessTokenEnabled()) {
+            jwtClaimsSetBuilder.claim(APP_DOMAIN, oAuthAppDO.getAppOwner().getTenantDomain());
```

Comment:
oAuthAppDO.getAppOwner().getTenantDomain() this seems to be resolving the tenant domain of the app through the appowners tenant domain. lets directly resolve the apps tenant domain.

Reply:
use app's tenant domain defined in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2384/files#diff-070f3f0502b6c65ce0efc4561caec88b5615025bc7da1ccc32d1e58f10da7654R456